meilisearch-http/src/extractors/authentication/mod.rs

```diff
@@ -126,15 +141,82 @@ pub mod policies {
                 return Some(AuthFilter::default());
             }
 
-            // authenticate if token is allowed.
-            if let Some(action) = Action::from_repr(A) {
+            if token.contains('.') {
+                // Tenant token
+                return ActionPolicy::<A>::authenticate_tenant_token(auth, token, index);
```

Comment:
why is that sufficient?

Reply:
API keys can't contain `.`, the master key could but it's checked before. We can change this into:
```suggestion
            if let Some(filters) = ActionPolicy::<A>::authenticate_tenant_token(auth, token, index) {
                // Tenant token
                return Some(filters);
```

which will try to authenticate to know if it is a tenant token or not. 🙂